### PR TITLE
Feature/ras 1d

### PR DESCRIPTION
**Pull request by sray014**
Adds functions to export the following 1D Output data from HDF files as well as unit tests for each function:
  * `Cross Sections`
  * `River Reaches`
  * `Cross Section Velocity`
  * `Steady Flow Names`
  * `Cross Section Area`
  * `Encroachment Information`
  * `Cross Section Elevation`
  * `Cross Section WSEL Information`
  * `Cross Section Energy Grade`
  * `Cross Section Flow`

### CODE REVIEW

**Review by slawler — CHANGES_REQUESTED** [review]
Test coverage is <85%. Please add additional tests as needed, and verify formatting is also passing.

**Comment by sduncan4** [comment]
@sray014 and @slawler , I've made the adjustments to geom.py for the encroachments functions. Please review.

File: src/rashdf/geom.py
Comment (thwllms) [inline src/rashdf/geom.py:715]: 1. Methods pulling from `.p**.hdf` output files (i.e. `/Results/...` path) belong in `RasPlanHdf` rather than `RasGeomHdf`, since the latter is meant for opening `.g**.hdf` files.
3.  Noticing that these methods all have nearly identical code:
* `cross_sections_encroachment_station_left`
* `cross_sections_encroachment_station_right`
* `cross_sections_area`
* `cross_sections_velocity`

Could make this simpler by having a single private function to retrieve output data and an enum with the names of all the variables.

```python
class XsSteadyOutput(Enum):
    ALPHA = "Alpha"
    AREA_FLOW_CHANNEL = "Area Flow Channel"
    AREA_FLOW_LEFT_OB = "Area Flow Left OB"
    AREA_FLOW_RIGHT_OB = "Area Flow Right OB"
    ...
    WETTED_PERIMETER_TOTAL = "Wetted Perimeter Total"

DEFAULT_XS_STEADY_OUTPUT = [
    XsSteadyOutputVar.ENERGY_GRADE,
    XsSteadyOutputVar.FLOW,
    XsSteadyOutputVar.WATER_SURFACE,
]

class RasPlanHdf(RasGeomHdf):
    ...
    def steady_profile_xs_output(self, var: XsSteadyOutputVar, round_to: int = 2) -> DataFrame
        if var in DEFAULT_XS_STEADY_OUTPUT:
            path = f"{self.STEADY_OUTPUT_XS}/{var.value}"
        else:
            path = f"{self.STEADY_OUTPUT_XS_ADDITIONAL_VARS}/{var.value}"
        if path not in self:
            return pd.DataFrame()

        profiles = self.steady_flow_names()

        enc_data = self[path]
        df = pd.DataFrame(enc_data, index=profiles)
        df_t = df.T.copy()
        for p in profiles:
            df_t[p] = df_t[p].apply(lambda x: round(x, round_to))

        return df_t
```

Now variable-specific methods can be added without code duplication:
```python
class RasPlanHdf(RasGeomHdf):
    ...
    def cross_sections_wsel(self) -> DataFrame:
        return self.steady_profile_xs_output(XsSteadyOutput.WATER_SURFACE)

    def cross_sections_flow(self) -> DataFrame:
        return self.steady_profile_xs_output(XsSteadyOutput.FLOW)

    def cross_sections_energy_grade(self) -> DataFrame:
        return self.steady_profile_xs_output(XsSteadyOutput.ENERGY_GRADE)

    def cross_sections_additional_output(self) -> DataFrame:
        df = None
        for var in XsSteadyOutput:
            if var not in DEFAULT_XS_STEADY_OUTPUT:
                df_output = self.steady_profile_xs_output(var).add_suffix(f" {var.value}")
                if df is None:
                    df = df_output
                else:
                    df = df.merge(df_output)  # not sure if this is the correct merge syntax
        return df
```

File: src/rashdf/geom.py
Comment (thwllms) [inline src/rashdf/geom.py:648]: Re: `.round(2).astype(str)` -- are there issues occurring if bank stations aren't rounded to 2 places and converted to string?

File: src/rashdf/geom.py
Comment (thwllms) [inline src/rashdf/geom.py:586]: We've been trying to stick long, oft-reused strings like this one class-level variables similar to `RasGeomHdf.GEOM_STRUCTURES_PATH`, `RasGeomHdf.FLOW_AREA_2D_PATH`, to help avoid misspelled paths, let our editors tab-complete the paths, make the things easier to modify in the future, etc.

File: src/rashdf/geom.py
Comment (thwllms) [reply src/rashdf/geom.py:715]: ☝ Not trying to be prescriptive, btw, just pointing out an existing strategy in the codebase. I'm all for better ways to avoid code duplication or strong arguments in favor of it for specific cases.

File: src/rashdf/geom.py
Comment (sduncan4) [reply src/rashdf/geom.py:586]: This was very helpful! You'll notice that I've updated the paths to follow your recommendations in the plan hdf, where the paths get long. I followed the examples you provided and the code already there.

File: src/rashdf/geom.py
Comment (sduncan4) [reply src/rashdf/geom.py:715]: I'm in favor of doing so as well. I appreciate the instruction and example. You should see the update in the plan file following the path classes and functions for the steady results. You're suggested draft was very helpful. I'd like to keep things simple and in the spirit of the other code written. Works fantastic too!

File: src/rashdf/geom.py
Comment (sduncan4) [reply src/rashdf/geom.py:648]: There are some instances (not always, but sometimes) where the floats in the hdf are precise in the banks and imprecise in the elevation data and vice versa. Most cases models only use 2 decimal places.

I like the round_to variable though for people to adjust if needed. Applying it in the xs elevations function as well

**Comment by sray014** [comment]
@thwllms I assume the version in the pyproject.toml will need to be bumped up to 0.3.1?

**Review by slawler — CHANGES_REQUESTED** [review]
@sray014 please update the version in the project.toml

**Review by slawler — APPROVED** [review]

**Comment by slawler** [comment]
@thwllms Looks like the issues have been resolved, I approved, will wait for you to merge once you've had a chance to review.

File: src/rashdf/plan.py
Comment (thwllms) [inline src/rashdf/plan.py:57]: * The `XS_STEADY_OUTPUT` and `XS_STEADY_ADDITIONAL` constants here are redundant since they both contain the full list of variables in each corresponding enum. The other lists of specific enum values above were created because there are places where only some of the enum values are relevant. You could either combine the enums into a single `XsSteadyOutputVar` enum and create an `XS_STEADY_OUTPUT_ADDITIONAL` list that contains only "additional" vars, or simply remove the `XS_STEADY_OUTPUT` and `XS_STEADY_ADDITIONAL` vars.

**Review by thwllms — APPROVED** [review]
LGTM, just the one comment about the enums/constants in `plan.py`.

**Comment by sray014** [comment]
@thwllms does fiona need to be added to the pyproject dependencies for those checks to work? Not sure why it worked on the previous commit and not this one since nothing changed that deals with that.

**Comment by thwllms** [comment]
> @thwllms does fiona need to be added to the pyproject dependencies for those checks to work? Not sure why it worked on the previous commit and not this one since nothing changed that deals with that.

Looks like this is happening because GeoPandas 1.0.0 was just released a few hours ago and I guess it doesn't rely on Fiona as an explicit dependency anymore. We were previously relying on version ~0.14. For now let's pin GeoPandas to version number 0.14.4 and create an issue to upgrade to 1.0.0.

In `pyproject.toml`:
```toml
name = "rashdf"
description = "Read data from HEC-RAS HDF files."
readme = "README.md"
classifiers = [
    "Development Status :: 4 - Beta",
    "Intended Audience :: Developers",
    "License :: OSI Approved :: MIT License",
    "Programming Language :: Python :: 3",
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
    "Programming Language :: Python :: 3.12",
]
version = "0.4.0"
dependencies = ["h5py", "geopandas>=0.14,<0.15", "pyarrow", "xarray"]
```


https://pypi.org/project/geopandas/#history

**Pull request merged**
